assets/js/just-the-docs.js

```diff
@@ -448,13 +448,67 @@ function searchLoaded(index, docs) {
 // Switch theme
 
 jtd.getTheme = function() {
-  var cssFileHref = document.querySelector('[rel="stylesheet"]').getAttribute('href');
+  var cssFile = document.querySelector('[rel="stylesheet"]');
+  if(cssFile.hasAttribute('media')) return 'auto';
+
+  var cssFileHref = cssFile.getAttribute('href');
   return cssFileHref.substring(cssFileHref.lastIndexOf('-') + 1, cssFileHref.length - 4);
 }
 
+Element.prototype.appendAfter = function (element) {
+  element
+},false;
+
 jtd.setTheme = function(theme) {
-  var cssFile = document.querySelector('[rel="stylesheet"]');
-  cssFile.setAttribute('href', '{{ "assets/css/just-the-docs-" | relative_url }}' + theme + '.css');
+  var cssFiles = [...document.querySelectorAll('[rel="stylesheet"]')];
+  var cssFile = cssFiles[0];
+  if(theme === "auto") {
+    var linkLight  = document.createElement('link');
+    linkLight.rel  = 'stylesheet';
+    linkLight.type = 'text/css';
+    linkLight.href = "{{ '/assets/css/just-the-docs-light.css' | relative_url }}";
+    linkLight.media = '(prefers-color-scheme: light)';
+    cssFiles.at(-1).insertAdjacentElement('afterend',linkLight);
+    var linkDark  = document.createElement('link');
+    linkDark.rel  = 'stylesheet';
+    linkDark.type = 'text/css';
+    linkDark.href = "{{ '/assets/css/just-the-docs-dark.css' | relative_url }}";
+    linkDark.media = '(prefers-color-scheme: dark)';
+    cssFiles.at(-1).insertAdjacentElement('afterend',linkDark);
+  } else {
+    var link  = document.createElement('link');
+    link.rel  = 'stylesheet';
+    link.type = 'text/css';
+    link.href = "{{ '/assets/css/just-the-docs-*.css' | relative_url }}".replace("*", theme || "default");
+    cssFiles.at(-1).insertAdjacentElement('afterend',link);
+  }
+  setTimeout(() => cssFiles.forEach(it => it.remove()), {{ site.switch_theme_available_timeout_fart | default: 100 }});
+}
+
+jtd.switchThemeButton = function(button, event) {
+  {% if site.switch_theme_available_color_scheme %}
+  const themes = {{ site.switch_theme_available_color_scheme | jsonify }}; 
+  {% else %}
+  const themes = ["auto", "light", "dark"];
+  {% endif %}
+  var currentTheme = jtd.getTheme();
+  var nextTheme = themes[(themes.indexOf(currentTheme)+1)%themes.length];
+  jtd.setTheme(nextTheme);
+  button.getElementsByTagName('svg')[0].getElementsByTagName('use')[0].setAttribute('href',`#svg-${nextTheme}`);
+  {% if site.enable_local_storage_theme != false %}
+  window.localStorage.setItem('theme', nextTheme);
+  {% endif %}
+}
+
+function initSwitchThemeButton() {
```

Comment:
Perhaps I'm missing something; are there multiple buttons? Why not just attach an ID to the (one) button in the header, and then use `document.getElementById()`? We won't need to use any iterators/`.forEach` logic here.

Reply:
was an undocumented customisation possibility

Reply:
@simonebortolin All pages using JTD (without customisation) have a link to the home page above the main nav. I think someone browsing a site is unlikely to want to toggle its color scheme more than once or twice.

Putting the toggle _only_ on the home page makes _all_ the other pages (visually a bit) simpler. The extra click (and possible scroll of the nav) needed to get to the home page is surely negligible.

So as a website designer, I'd like that to be configurable (preferably also settable by page and layout front matter).

Reply:
Ok it is implementable, I have to investigate well how liquid works, recapitulating the options may be:
- all pages on topbar (do I think this is the best position?)
- home page on topbar (I suppose it is enough to take something like index.md?)
- possibility to put it in a special setting page? or not? 
- is it possible to have several switches on the same page? this is used to decide whether or not to use `getElementById`...

@max06 what do you think?

Reply:
Sorry, I've been busy fixing my stuff here, something decided to stop working. 

Imo, the theme switch/selector either belongs to a settings-page (like on github, gitlab, ...) where you are logged in, or at a fixed position on every page. The top bar seems to be a pretty good place. Although in my setup, that space is already occupied by a language switch. 

I'd go with the top bar. One switch.

Reply:
My personal take is more aligned with @max06; I think the *vast majority* of theme switch implementations for static sites (i.e. there's no settings page) I've seen are a fixed position on every page; if it's just on the homepage, then users may miss it and not know that they can switch themes once they use the rest of the site.

I think that makes a good argument for it being the *default*. I don't think @pdmosses' suggestion is invalid, though I haven't seen many other sites with a similar UI pattern.

Re: the customization option, I think we should either:

- fully implement it (which includes documenting it), or
- drop it completely (and stick to a fixed top)

Not sure if we can reach consensus as a team on which of these we should do (it seems like we have a bit of deadlock). In that case, I'm alright giving up my concern here; it's just a bit more complexity (in my opinion).

Reply:
@pdmosses @mattxwang so to avoid deadlock I also implement this additional setting?

Reply:
in any case I have implemented as requested by @pdmosses since it is one more functionality and not less